enterprise/api_client/discovery.py

```diff
@@ -377,13 +383,16 @@ def _load_data(self, resource, default=DEFAULT_VALUE_SAFEGUARD, **kwargs):
 
         """
         default_val = default if default != self.DEFAULT_VALUE_SAFEGUARD else {}
-        result = get_edx_api_data(
-            api_config=CatalogIntegration.current(),
-            resource=resource,
-            api=self.client,
-            **kwargs
-        )
-        return result or default_val
+        try:
+            return get_edx_api_data(
+                api_config=CatalogIntegration.current(),
+                resource=resource,
+                api=self.client,
+                **kwargs
+            ) or default_val
+        except HttpClientError as exc:
+            LOGGER.error('Failed to load data from resource %s with kwargs %s due to: %s', resource, kwargs, str(exc))
```

Comment:
Can we change these to `LOGGER.exception` calls instead so we get the full stack trace in the logs?

Reply:
Yes, sure!